llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1632,6 +1632,16 @@ Instruction *InstCombinerImpl::visitUDiv(BinaryOperator &I) {
         I, Builder.CreateLShr(Op0, Res, I.getName(), I.isExact()));
   }
 
+  // Op0 udiv Op1 -> Op0 lshr cttz(Op1), if Op1 is a power of 2.
+  if (isKnownToBeAPowerOfTwo(Op1, /*OrZero*/ true, /*Depth*/ 0, &I)) {
+    // This will increase instruction count but it's okay
+    // since bitwise operations are substantially faster than
+    // division.
```

Comment:
IIRC udiv with a power of 2 divisor only takes 4 cycles on cortex-a53. Therefore this transformation is not always a win. Can you double-check this behavior, if possible?